SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/LegacySpatialWorkerConnection.cpp

```diff
@@ -95,31 +96,36 @@ Worker_RequestId ULegacySpatialWorkerConnection::SendReserveEntityIdsRequest(uin
 }
 
 Worker_RequestId ULegacySpatialWorkerConnection::SendCreateEntityRequest(TArray<FWorkerComponentData> Components,
-																		 const Worker_EntityId* EntityId)
+																		 const Worker_EntityId* EntityId,
+																		 const TOptional<worker::c::Trace_SpanId>& SpanId)
 {
-	QueueOutgoingMessage<FCreateEntityRequest>(MoveTemp(Components), EntityId);
+	QueueOutgoingMessage<FCreateEntityRequest>(MoveTemp(Components), EntityId, SpanId);
 	return NextRequestId++;
 }
 
-Worker_RequestId ULegacySpatialWorkerConnection::SendDeleteEntityRequest(Worker_EntityId EntityId)
+Worker_RequestId ULegacySpatialWorkerConnection::SendDeleteEntityRequest(Worker_EntityId EntityId,
+																		 const TOptional<worker::c::Trace_SpanId>& SpanId)
 {
-	QueueOutgoingMessage<FDeleteEntityRequest>(EntityId);
+	QueueOutgoingMessage<FDeleteEntityRequest>(EntityId, SpanId);
 	return NextRequestId++;
 }
 
-void ULegacySpatialWorkerConnection::SendAddComponent(Worker_EntityId EntityId, FWorkerComponentData* ComponentData)
+void ULegacySpatialWorkerConnection::SendAddComponent(Worker_EntityId EntityId, FWorkerComponentData* ComponentData,
+													  const TOptional<worker::c::Trace_SpanId>& SpanId)
 {
-	QueueOutgoingMessage<FAddComponent>(EntityId, *ComponentData);
+	QueueOutgoingMessage<FAddComponent>(EntityId, *ComponentData, SpanId);
 }
 
-void ULegacySpatialWorkerConnection::SendRemoveComponent(Worker_EntityId EntityId, Worker_ComponentId ComponentId)
+void ULegacySpatialWorkerConnection::SendRemoveComponent(Worker_EntityId EntityId, Worker_ComponentId ComponentId,
+														 const TOptional<worker::c::Trace_SpanId>& SpanId)
 {
-	QueueOutgoingMessage<FRemoveComponent>(EntityId, ComponentId);
+	QueueOutgoingMessage<FRemoveComponent>(EntityId, ComponentId, SpanId);
 }
 
-void ULegacySpatialWorkerConnection::SendComponentUpdate(Worker_EntityId EntityId, FWorkerComponentUpdate* ComponentUpdate)
+void ULegacySpatialWorkerConnection::SendComponentUpdate(Worker_EntityId EntityId, FWorkerComponentUpdate* ComponentUpdate,
+														 const TOptional<worker::c::Trace_SpanId>& SpanId)
 {
-	QueueOutgoingMessage<FComponentUpdate>(EntityId, *ComponentUpdate);
+	QueueOutgoingMessage<FComponentUpdate>(EntityId, *ComponentUpdate, SpanId);
 }
 
 Worker_RequestId ULegacySpatialWorkerConnection::SendCommandRequest(Worker_EntityId EntityId, Worker_CommandRequest* Request,
```

Comment:
Why do we not trace sending command requests?

Reply:
I guess it's just not part of the specification. Only certain events where requested. We could definitely expand and add this in future.

FYI: list of events specified here: https://improbableio.atlassian.net/browse/UNR-3964

Reply:
I think we should instrument sending command requests as well since they're used for the cross server RPCs. I've left a comment in the ticket as well.